backend-shared/util/chaosclient.go

```diff
@@ -0,0 +1,196 @@
+}
+
+// Update updates the given obj in the Kubernetes cluster. obj must be a
+// struct pointer so that obj can be updated with the content returned by the Server.
+func (pc *ChaosClient) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	res := pc.InnerClient.Update(ctx, obj, opts...)
+
+	if err := shouldSimulateFailure(); err != nil {
+		return err
+	}
+
+	return res
+}
+
+// Patch patches the given obj in the Kubernetes cluster. obj must be a
+// struct pointer so that obj can be updated with the content returned by the Server.
+func (pc *ChaosClient) Patch(ctx context.Context, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+	res := pc.InnerClient.Patch(ctx, obj, patch, opts...)
+
+	if err := shouldSimulateFailure(); err != nil {
+		return err
+	}
+
+	return res
+}
+
+// DeleteAllOf deletes all objects of the given type matching the given options.
+func (pc *ChaosClient) DeleteAllOf(ctx context.Context, obj client.Object, opts ...client.DeleteAllOfOption) error {
+	res := pc.InnerClient.DeleteAllOf(ctx, obj, opts...)
+
+	if err := shouldSimulateFailure(); err != nil {
+		return err
+	}
+
+	return res
+}
+
+// StatusWriter knows how to update status subresource of a Kubernetes object.
+func (pc *ChaosClient) Status() client.StatusWriter {
+	res := pc.InnerClient.Status()
+	return res
+}
+
+// Scheme returns the scheme this client is using.
+func (pc *ChaosClient) Scheme() *runtime.Scheme {
+	res := pc.InnerClient.Scheme()
+	return res
+
+}
+
+// RESTMapper returns the rest this client is using.
+func (pc *ChaosClient) RESTMapper() meta.RESTMapper {
+	res := pc.InnerClient.RESTMapper()
+	return res
+}
+
+type ChaosClientStatusWrapper struct {
+	innerWriter *client.StatusWriter
+	// parent      *ProxyClient
+}
+
+// Update updates the fields corresponding to the status subresource for the
+// given obj. obj must be a struct pointer so that obj can be updated
+// with the content returned by the Server.
+func (pcsw *ChaosClientStatusWrapper) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+
+	res := (*pcsw.innerWriter).Update(ctx, obj, opts...)
+
+	if err := shouldSimulateFailure(); err != nil {
+		return err
+	}
+
+	return res
+
+}
+
+// Patch patches the given object's subresource. obj must be a struct
+// pointer so that obj can be updated with the content returned by the
+// Server.
+func (pcsw *ChaosClientStatusWrapper) Patch(ctx context.Context, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+
+	if err := shouldSimulateFailure(); err != nil {
+		return err
+	}
+
+	res := (*pcsw.innerWriter).Patch(ctx, obj, patch, opts...)
+
+	return res
+}
+
+func shouldSimulateFailure() error {
+
+	if !isEnvExist("UNRELIABLE_CLIENT_FAILURE_RATE") {
+		return nil
+	}
+
+	unreliableClientFailureRate := os.Getenv("UNRELIABLE_CLIENT_FAILURE_RATE")
+	unreliableClientFailureRateValue, err := strconv.Atoi(unreliableClientFailureRate)
```

Comment:
@Rizwana777 , following up from standup, try this:
```go
	unreliableClientFailureRate := os.Getenv("UNRELIABLE_CLIENT_FAILURE_RATE")
	unreliableClientFailureRateValue, err := strconv.Atoi(unreliableClientFailureRate)
	if err != nil {
		return err
	}
	
	// Convert it to a decimal point, e.g. 50% to 0.5 
	unreliableClientFailureRateValue = unreliableClientFailureRateValue / 100
	
	// return an error randomly x% of the time
	if rand.Float64() < unreliableClientFailureRateValue {
		return fmt.Errorf("simulated K8s error")
	}
	
	return nil
```

Reply:
Thanks @jgwest